jgrapht-core/src/main/java/org/jgrapht/alg/intervalgraph/LexBreathFirstSearch.java

```diff
@@ -0,0 +1,217 @@
+        
+        for(int i = n; i >= 1; i--) {
+            // pick unvisited vertex v with largest label
+            ArrayList<V> candidates = labelToVertexMap.get(maxLabel);
+            V currentVertex = candidates.get(0);
+            
+            // set index of v to be n + 1 - i
+            ordering.add(currentVertex);
+            
+            // for every unvisited neighbor w of v, append i to label of w
+            for(V neighbor : Graphs.neighborListOf(graph, currentVertex)) {
+                String oldLabel = vertexToLabelMap.get(neighbor);
+                String newLabel = oldLabel + Integer.toString(i);
+                
+                // remove neighbor from previous list
+                ArrayList<V> oldList = labelToVertexMap.get(oldLabel);
+                oldList.remove(neighbor);
+                
+                // if there are no vertices left in this list, delete the entry
+                if(oldList.isEmpty()) {
+                    labelToVertexMap.remove(oldLabel);
+                } else {
+                    // else we replace the list with its updated version 
+                    labelToVertexMap.replace(oldLabel, oldList);
+                }
+                
+                // now add neighbor to the list of the new label
+                if(labelToVertexMap.containsKey(newLabel)) {
+                    ArrayList<V> newList = labelToVertexMap.get(newLabel);
+                    newList.add(neighbor);
+                    labelToVertexMap.replace(newLabel, newList);
+                } else {
+                    ArrayList<V> newList = new ArrayList<V>();
+                    newList.add(neighbor);
+                    labelToVertexMap.put(newLabel, newList);
+                }
+                
+                // associate neighbor with its new label
+                vertexToLabelMap.replace(neighbor, newLabel);
+            }
+            
+            // mark current vertex as visited, i.e. remove its entries from the maps
+            vertexToLabelMap.remove(currentVertex);
+            candidates.remove(currentVertex);
+            labelToVertexMap.replace(maxLabel, candidates);
+        }
+    
+        // return ordering -- thanks Java for not being able to create an array from a list with generic types for some reason
+        V[] orderingArray = null;
+        orderingArray = ordering.toArray(orderingArray);
+        
+        return orderingArray;
+    }
+    
+    /**
+     * Performs LBFS+ starting at {@code startingVertex} using the previous ordering {@code prevOrdering}.
+     *
+     * @param graph the graph we want to perform LBFS on
+     * @param startingVertex the starting vertex of the LBFS
+     * @param prevOrdering the previous LBFS ordering given as an array
+     * @return an array of vertices representing the order in which the vertices were found
+     */
+    public V[] lexBreathFirstSearch(Graph<V, E> graph, V startingVertex, V[] prevOrdering)
```

Comment:
This is a lot of copypasta code. Maybe change this to allow prevOrdering to be null which would turn off the sorting (which is the only difference in this algorithm?)

Reply:
Im not adding any more comments to this algorithm as it is the same as the one above.